File: scenarios/telemetry/dotnet/telemetry_producer/PositionTelemetryProducer.cs
using GeoJSON.Text.Geometry;
using MQTTnet.Client;
using MQTTnet.Client.Extensions;

namespace telemetry_producer;

internal class PositionTelemetryProducer : TelemetryProducer<Point>
{
    public PositionTelemetryProducer(IMqttClient mqttClient)
        : base(mqttClient, new Utf8JsonSerializer(), "vehicles/{clientId}/position")
    {
    }
}

File: scenarios/telemetry/ts/telemetryProducer/src/positionTelemetryProducer.ts
import {
    MqttClient
} from 'mqtt';
import {
    GeoJsonPoint,
    Utf8JsonSerializer,
    TelemetryProducer,
} from '@mqttapplicationsamples/mqttjsclientextensions';

export class PositionTelemetryProducer extends TelemetryProducer<GeoJsonPoint> {
    constructor(mqttClient: MqttClient) {
        super(mqttClient, new Utf8JsonSerializer(), "vehicles/{clientId}/position");
    }
}
